torchci/components/MultiSelectPicker.tsx

```diff
@@ -0,0 +1,114 @@
+// a component that has a list of options and allows the user to select multiple options, with a 'select all' and 'clear' button.
+
+import {
+  Button,
+  Checkbox,
+  FormControl,
+  InputLabel,
+  MenuItem,
+  Select,
+  SelectChangeEvent,
+} from "@mui/material";
+import { useState } from "react";
+
+export default function MultiSelectPicker({
+  selected,
+  setSelected,
+  options,
+  label,
+  renderValue,
+  style,
+}: {
+  selected: string[] | undefined;
+  setSelected: any;
+  options: string[];
+  label: string;
+  renderValue: (selected: string[]) => string;
+  style?: { [key: string]: any };
+}) {
+  var [selectedItems, setSelectedItems] = useState(selected ?? []);
+
+  console.log("incoming options", options);
+
+  function handleChange(e: SelectChangeEvent<string[]>) {
+    const newList = e.target.value;
+
+    console.log(`newList: ${newList}`);
+
+    // if the last item is empty, a button was pressed and we skip the update
+    if (newList.length > 0 && !newList[newList.length - 1]) {
+      return;
+    }
+    // if newList is a string, skip the update
+    if (typeof newList === "string") {
+      console.log("newList is a string - skipping update");
+      return;
+    }
+    setSelectedItems(newList);
+    setSelected(newList);
+    // }
+  }
+
+  function selectAll() {
+    setSelectedItems(options);
+    setSelected(options);
+  }
+
+  function clear() {
+    setSelectedItems([]);
+    setSelected([]);
```

Comment:
I'm getting an error when I clear, select one, then clear again, not sure where it originates from tho
```
cost_analysis-5c15b12c3c5cb750.js:1 Uncaught TypeError: Cannot read properties of undefined (reading 'stack')
    at formatter (cost_analysis-5c15b12c3c5cb750.js:1:4899)
    at e._showTooltipContent (2578-8510c40faee19ce3.js:1:879594)
    at e.<anonymous> (2578-8510c40faee19ce3.js:1:878278)
    at e._showOrMove (2578-8510c40faee19ce3.js:1:876240)
    at e._showSeriesItemTooltip (2578-8510c40faee19ce3.js:1:878247)
    at e._tryShow (2578-8510c40faee19ce3.js:1:875961)
    at e.<anonymous> (2578-8510c40faee19ce3.js:1:873090)
    at _V (2578-8510c40faee19ce3.js:1:782720)
    at 2578-8510c40faee19ce3.js:1:16939
    at 2578-8510c40faee19ce3.js:1:782303
```

Reply:
fixed